src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py

```diff
@@ -998,12 +998,13 @@ def call(
         pred_hid = last_hidden[:, -tgt_len:]
 
         softmax_output = self.crit(pred_hid, labels, training=training)
+        prediction_scores = softmax_output if labels is None else ()
```

Comment:
An empty tuple? Is that really what's expected? Seems weird.

Reply:
This is done in PyTorch version. I am not sure about the reason why not to output `prediction_scores` (empty tuple).

`softmax_output` is returned by [ProjectedAdaptiveLogSoftmax](https://github.com/huggingface/transformers/blob/6f9d8dc1567889b073b91059c9b19e9a6813abfa/src/transformers/models/transfo_xl/modeling_transfo_xl_utilities.py#L29), which is quite complicated. In particular, when `labels` is passed to `ProjectedAdaptiveLogSoftmax`, the logic is very different from not passing `labels`.

Reply:
Oh, there is one reason doing so in PyTorch version:

```
 softmax_output.view(bsz, tgt_len, -1)
```
This shape only works when `labels` is not used.